Refactored Datasource and RDD

helipilot50 opened the pull request:
Hi,

I have completed a complete refactoring of the connector.

New features:
- Read and write a DataFrame
- Spark SQL filters pushed down to Aerospike servers
- AerospikeClient cache
- Efficient RDD partitioning, reducing cost of creating AerospikeClien instances

helipilot50 committed:
Added a client cache to avoid the cost of getting a new client on each query on each worker and each task

helipilot50 committed:
Initial work on read-write function and integration of QueryEngine

helipilot50 committed:
Initial work on a new Aerospike RDD with Read/Write package functions

helipilot50 committed:
Added aggragation by node
Added unit test

helipilot50 committed:
QueryEngine tests

helipilot50 committed:
DefaultSource and Relation tests

helipilot50 committed:
Implemented row iteration on resultset

helipilot50 committed:
Debugging AerospikeRelation

helipilot50 committed:
Added code to save DataFrame
Debugged type conversion -- more work required

helipilot50 committed:
Successful completion of "scan" read of Aerospike set into DataFrame

helipilot50 committed:
added configurable metadata column names
added tests for AerospikeConfig
start work on 'save'

helipilot50 committed:
More work on saving DataFrame

helipilot50 committed:
Implemented write policy features in Config
Added elementaty support for maps
Testes filters successfully

helipilot50 committed:
Started README
added assembly

helipilot50 committed:
Updated build
Added assembly packaging

helipilot50 committed:
Update README.md

helipilot50 committed:
Added test for saving DataFrame
Corrected digest and key columns
Added initial support for list and map

helipilot50 committed:
improving assembly
added better test for scan

helipilot50 committed:
added tests in parallel

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
removed unnecessary code

helipilot50 committed:
Merge commit '519e20c4225ced7806139f872e73e3928fa81e91' into Read-Write-1.0

* commit '519e20c4225ced7806139f872e73e3928fa81e91':
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md

helipilot50 committed:
excluded Spark libraries from the uber JAR

Added more info in the README

helipilot50 committed:
Added more filters
Added javadoc

helipilot50 committed:
Merge pull request #1 from helipilot50/master

Merged to master

helipilot50 committed:
ignoring metastore

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Merge pull request #1 from helipilot50/master

Testing and documentation

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Merge pull request #2 from helipilot50/master

more documentation testing

helipilot50 committed:
Update README.md

helipilot50 committed:
Merge pull request #3 from helipilot50/master

Update README.md

helipilot50 committed:
Update README.md

helipilot50 committed:
Merge pull request #4 from helipilot50/master

fixed url

thefishermanhacker commented:
Looks good, i'll test it once it will be commit!
Thanks!!!

sasha-polev commented:
Hi. For variety of reasons it is not too likely this pull request will be merged in this repo, but you can check it out here: https://github.com/aerospike/aerospark

helipilot50 committed:
Fixed defect in aerospike-helper
Updated build script

helipilot50 committed:
Added support for new types
Added TTL support on save
Enhanced error messages

helipilot50 committed:
Merge pull request #5 from helipilot50/master

New types and TTL support on save

helipilot50 committed:
Added test data
Included helper class JAR

helipilot50 committed:
Improvements in parallelism and logging

helipilot50 committed:
Added timeout support
improved default value processing

helipilot50 committed:
Added a vagrant cluster definition for testing -- requires Vagrant and Virtual Box
Improved all tests
Refactored field mapping to be more efficient
Fixed bug in fields required in SQL statement

wchu-citrusleaf committed:
Merge pull request #6 from helipilot50/master

Improvement in SQL column support

helipilot50 committed:
Removed commented out code
Updated documentation

helipilot50 committed:
Added save mode table

wchu-citrusleaf committed:
Merge pull request #7 from helipilot50/master

Improved documentation

helipilot50 committed:
Added support to read and write Lists

helipilot50 committed:
Implemented Map support

wchu-citrusleaf committed:
Merge pull request #8 from helipilot50/master

Support for Reading/Writing Lists and Maps

helipilot50 committed:
Refactored tests to use global values

helipilot50 committed:
Refactored tests to use global values

helipilot50 committed:
Merge branch 'master' of https://github.com/helipilot50/aerospike-spark

helipilot50 committed:
less verbose log messages in test

wchu-citrusleaf committed:
Merge pull request #9 from helipilot50/master

Refactored test to use global variables

snadorp committed:
Upgrade to Spark 2.0 (#10)

* Format the code.

* Remove unnecessary variable shadowing.

* Code clean up.

* Format the test code.

* Refactor tests.

* Upgrade to Spark 2.0.

Commit:
Fix build for ubuntu and update README.

Commit:
Changing Append SaveMode to RecordExistsAction.UPDATE (#11)

Commit:
Fix the aerospark git link

Commit:
One more aerospark link fix in README

Commit:
Added Binary type to TypeConverter.binToValue

mzhangsfly committed:
Merge pull request #13 from eluppol/master

Added Binary type to TypeConverter.binToValue

mzhangsfly committed:
Added join/intersect functionality, updated test to use docker, removed helper jar from source (#15)

* - Replaced the aerospike-helper jar with maven configuration instead
- Added Dataset methods for join operation
- Added docker container for testing instead of hardcoded test server
host
- Aerospace client cache enhancement with socket timeout support
- Client cache bug fix
- Added AND/OR aerospike query support

* updated the spark lib as provided

* Added the socket time out config

* Code re-org to use default scala structure for testing

* Made  the container as a singleton object

* Rename spec file

* Updated readme

* Added large data save/read back

* Put the default aerospike host back to 127.0.0.1

* Removed publish to local from build

Commit:
Wait 5 seconds for AS container to boot up

Commit:
Clean test log output

mzhangsfly committed:
Clean test log output (#16)

Commit:
Added scanSet function for SparkSession
Added save function and some tune-up for Dataset
Added concurrent save for better performance with configurable number
of threads
And more tests

Commit:
Added wait for all threads complete

Commit:
Added aeroBatchRead function

Commit:
added support to provide bin name array to batch.get to reduce network load

Commit:
Dataset save and join tune-up

Added function of setName: aerospike set name, and key: object
to-be-saved key field
aeroJoin to return a Map of bins

Commit:
Added aeroRead function to spark session

Commit:
Code clean up

Commit:
Version to 1.3

Commit:
Update version to 1.3

Commit:
Utilize the runtime config instead of sparkConf for areospike connection

Commit:
Filter null records

Commit:
Clean up

Commit:
Merge branch 'master' of https://github.com/mzhangsfly/aerospark

mzhangsfly committed:
Join and Intersect functionality (#17)

* Clean test log output

* Added scanSet function for SparkSession
Added save function and some tune-up for Dataset
Added concurrent save for better performance with configurable number
of threads
And more tests

* Added wait for all threads complete

* Added aeroBatchRead function

* added support to provide bin name array to batch.get to reduce network load

* Dataset save and join tune-up

Added function of setName: aerospike set name, and key: object
to-be-saved key field
aeroJoin to return a Map of bins

* Code clean up

* Version to 1.3

* Utilize the runtime config instead of sparkConf for areospike connection

* Filter null records

* Clean up

Commit:
AeroJoin optimization

- added the concurrent batch read utilizing twitter Future/Promise
- renamed aeroJoin to aeroJoinMap, which returns a map of records
- updated aeroJoin function returns a strong typed dataset
- added some relevant tests
- version update to 1.3.1

Commit:
Merge branch 'master' of https://github.com/aerospike/aerospark

mzhangsfly committed:
Merge pull request #19 from mzhangsfly/master

aeroJoin optimization

jaceklaskowski committed:
[MINOR] "Upgrade" code snippets to Spark 2

jaceklaskowski committed:
Upgrade to Spark 2.1.1 (and remove spark-csv)

jaceklaskowski committed:
Define the sbt version in use in the project

mzhangsfly committed:
Merge pull request #22 from jaceklaskowski/build-spark211

Upgrade to Spark 2.1.1

mzhangsfly committed:
Merge pull request #20 from jaceklaskowski/readme

[MINOR] "Upgrade" code snippets to Spark 2

Commit:
adds BinaryType conversion to fieldToBin

mzhangsfly committed:
Merge pull request #23 from alexbelj/feature/binary_type_conversion

adds BinaryType conversion to fieldToBin

mzhangsfly committed:
aeroJoin optimization with parallelization (#24)

* Readme update and import clean up

* README update

* aeroJoin optimization with parallelization and test cleanup
